tutorials/migrating_ardupilot_plugin.md

```diff
@@ -0,0 +1,926 @@
+        ${IGNITION-COMMON_LIBRARIES}
+        ${IGNITION-GAZEBO_LIBRARIES}
+        ${IGNITION-MATH_LIBRARIES}
+        ${IGNITION-MSGS_LIBRARIES}
+        ${IGNITION-PHYSICS_LIBRARIES}
+        ${IGNITION-SENSORS_LIBRARIES}
+        ${IGNITION-TRANSPORT_LIBRARIES}
+        )
+```
+
+## The model
+
+The old UAV is defined in two parts: (i) the `iris_with_standoffs` model, which
+defines the vehicle structure; and (ii) the `iris_with_ardupilot` model, which
+includes the extends the `iris_with_standoffs` model by adding plugins needed to
+fly it.
+
+Because model inclusion is not (yet?) supported, the new model just combines
+the additional plugin configuration from `iris_with_ardupilot` into
+`iris_with_standoffs`. Along the way a few changes are made, as follows.
+
+The `<script>` tag for visual material is not (yet?) supported, so in the
+model, instances of `<script>` are just commented out (which leaves the UAV
+visually untextured, but functional).
+
+In the old model, loading an instance of the LiftDrag plugin for each half of
+each propeller looks like this:
+
+```xml
+<!-- OLD -->
+<plugin
+    name="rotor_0_blade_1"
+    filename="libLiftDragPlugin.so">
+  <!-- ...configuration goes here... -->
+  <link_name>iris::rotor_0</link_name>
+</plugin>
+```
+
+In the new model, we do this instead:
+
+```xml
+<!-- NEW -->
+<plugin
+    name="ignition::gazebo::systems::LiftDrag"
+    filename="libignition-gazebo3-lift-drag-system.so">
+  <!-- ...configuration goes here... -->
+  <link_name>rotor_0</link_name>
+</plugin>
+```
+
+In the old model, it's possible to read joint state and apply joint forces
+automatically. In the new model, we must instantiate the `JointStatePublisher`
+plugin once for the entire model and the `ApplyJointForce` plugin once for each propeller joint:
+
+```xml
+<!-- NEW -->
+<plugin
+  filename="libignition-gazebo-joint-state-publisher-system.so"
+  name="ignition::gazebo::systems::JointStatePublisher"></plugin>
+<plugin
+  filename="libignition-gazebo-apply-joint-force-system.so"
+  name="ignition::gazebo::systems::ApplyJointForce">
+  <joint_name>rotor_0_joint</joint_name>
+</plugin>
+<plugin
+  filename="libignition-gazebo-apply-joint-force-system.so"
+  name="ignition::gazebo::systems::ApplyJointForce">
+  <joint_name>rotor_1_joint</joint_name>
+</plugin>
+<plugin
+  filename="libignition-gazebo-apply-joint-force-system.so"
+  name="ignition::gazebo::systems::ApplyJointForce">
+  <joint_name>rotor_2_joint</joint_name>
+</plugin>
+<plugin
+  filename="libignition-gazebo-apply-joint-force-system.so"
+  name="ignition::gazebo::systems::ApplyJointForce">
+  <joint_name>rotor_3_joint</joint_name>
+</plugin>
+```
+
+## Building and running the plugin
+
+Now that you're familiar with the changes required by the migration from Gazebo
+to Ignition, you can try building and running the result of migration by
+following [these
+instructions](https://github.com/gerkey/ardupilot_gazebo/tree/ignition#using-with-ignition).
```

Comment:
This conclusion isn't really relevant (my mistake). I'd suggest making it clear that this is an aside from the main topic of migration, maybe like:

> If you were following along with the migration on your own fork of `ardupilot_gazebo` and would like to confirm that it does in fact now work on Ignition, you can build and run the result of the migration by following [these instructions](https://github.com/gerkey/ardupilot_gazebo/tree/ignition#using-with-ignition) (cloning your own repo in step 4)

Feels a little complicated though, especially for something most people wouldn't be doing anyway. Otherwise I'd suggest just removing any mention of building and running completely and say something like:

> You should be able to apply the same general changes covered in this tutorial to your Gazebo plugins to migrate them to Ignition.
>
> Check out [these instructions](https://github.com/gerkey/ardupilot_gazebo/tree/ignition#using-with-ignition) if you'd like to learn more about using `ardupilot_gazebo` with Ignition.

Just to mark a clear ending.

Reply:
Good idea. Done in 85f187a0eccfe18ca00b554dea30525667bef6ad